src/server/dfly_main.cc

```diff
@@ -161,9 +162,33 @@ struct VersionMonitor {
   }
 
  private:
+  bool IsVersionOutdated(std::string_view remote, std::string_view current) const;
   void RunTask(SSL_CTX* ssl_ctx);
 };
 
+bool VersionMonitor::IsVersionOutdated(const std::string_view remote,
+                                       const std::string_view current) const {
+  const std::vector<absl::string_view> remote_xyz = absl::StrSplit(remote, ".");
+  const std::vector<absl::string_view> current_xyz = absl::StrSplit(current, ".");
+  if (remote_xyz.size() != current_xyz.size()) {
+    LOG(WARNING) << "Remote and current version fields must be of equal size";
```

Comment:
Nit: I'd rephrase this message to something that can provide more context to users, who will see this printed out of nowhere.
Maybe: `"Can't compare Dragonfly version " << current << " to latest version " << remote;` ?